File: 0230) Kth Smallest Element in a BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> nums = new ArrayList<Integer>();
        helper(root, nums);
        Collections.sort(nums);
        return nums.get(k-1);
    }
    public void helper(TreeNode root, List<Integer> nums)
    {
        if(root == null)
        {
            return;
        }
        nums.add(root.val);
        helper(root.left, nums);
        helper(root.right, nums);
    }
}

File: 0230) Kth Smallest Element in a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        vector<int> nums;
        preorderTraversal(root, nums);
        sort(nums.begin(), nums.end());
        return nums[k-1];
    }
    void preorderTraversal(TreeNode *root, vector<int> &nums)
    {
        if(root == nullptr)
        {
            return;
        }
        nums.push_back(root->val);
        preorderTraversal(root->left, nums);
        preorderTraversal(root->right, nums);
    }
};
